jackson-databind/src/main/java/io/micronaut/jackson/databind/JacksonDatabindMapper.java

```diff
@@ -80,7 +84,14 @@ public ObjectMapper getObjectMapper() {
 
     @Override
     public <T> T readValueFromTree(@NonNull JsonNode tree, @NonNull Argument<T> type) throws IOException {
-        return objectMapper.readValue(treeCodec.treeAsTokens(tree), JacksonConfiguration.constructType(type, objectMapper.getTypeFactory()));
+        JsonParser tokens = treeCodec.treeAsTokens(tree);
+        JavaType javaType = JacksonConfiguration.constructType(type, objectMapper.getTypeFactory());
+        Optional<Class<?>> view = type.findAnnotation(JsonView.class).flatMap(AnnotationValue::classValue);
```

Comment:
Can you use `type.getAnnotationMetadat().classValue(JsonView.class)` instead. It will be more efficient as it doesn't have to create the whole annotation instance